lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,566 @@
+                elif key == "source_database_id":
+                    self.parameters["source_database_id"] = kwargs[key]
+                elif key == "source_database_deletion_date":
+                    self.parameters["source_database_deletion_date"] = kwargs[key]
+                elif key == "restore_point_in_time":
+                    self.parameters["restore_point_in_time"] = kwargs[key]
+                elif key == "recovery_services_recovery_point_resource_id":
+                    self.parameters["recovery_services_recovery_point_resource_id"] = kwargs[key]
+                elif key == "edition":
+                    self.parameters["edition"] = _snake_to_camel(kwargs[key], True)
+                elif key == "max_size_bytes":
+                    self.parameters["max_size_bytes"] = kwargs[key]
+                elif key == "requested_service_objective_id":
+                    self.parameters["requested_service_objective_id"] = kwargs[key]
+                elif key == "requested_service_objective_name":
+                    self.parameters["requested_service_objective_name"] = _snake_to_camel(kwargs[key], True)
+                elif key == "elastic_pool_name":
+                    self.parameters["elastic_pool_name"] = kwargs[key]
+                elif key == "read_scale":
+                    self.parameters["read_scale"] = 'Enabled' if kwargs[key] else 'Disabled'
+                elif key == "sample_name":
+                    ev = kwargs[key]
+                    if ev == 'adventure_works_lt':
+                        ev = 'AdventureWorksLT'
+                    self.parameters["sample_name"] = ev
+                elif key == "zone_redundant":
+                    self.parameters["zone_redundant"] = 'Enabled' if kwargs[key] else 'Disabled'
+
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        if "location" not in self.parameters:
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_sqldatabase()
+
+        if not old_response:
+            self.log("SQL Database instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("SQL Database instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                self.log("Need to check if SQL Database instance has to be deleted or may be updated")
+                if ('location' in self.parameters) and (self.parameters['location'] != old_response['location']):
```

Comment:
Can any of these values actually be changed on an existing database? It seems like they can only be set on creation time and you would need to delete the recreate the database, similar to what you did on the other database modules.

Reply:
in case of sql update actually works (only PostgreSQL and MySQL report internal error)